### PR TITLE
replace date input component

### LINKED ISSUE #1259: Date picker
Use Date picker in data entry instead of input field

https://projects.wojtekmaj.pl/react-date-picker/

### PR DESCRIPTION
**Pull request by ramirobg94**
resolves #1259

### CODE REVIEW

File: webapp/components/survey/SurveyForm/nodeDefs/components/types/nodeDefDate.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  const handleChangeDateInput = (newDate) => {
+    const newDateSplited = newDate.split('/')
+    if (newDateSplited.length === 3) {
+      updateNode(nodeDef, node, newDate)
+    }
+  }
+
+  return (
+    <div className="survey-form__node-def-date">
+      <DateInput
```
Comment (minotogna) [inline webapp/components/survey/SurveyForm/nodeDefs/components/types/nodeDefDate.js:30]: this component must be wrapped into a common component to be used also in other places

**Review by minotogna — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -142,6 +142,7 @@
     "react-avatar-editor": "^12.0.0-beta.0",
     "react-dom": "^17.0.1",
     "react-grid-layout": "^1.1.1",
+    "react-input-date": "^1.0.2",
```
Comment (SteRiccio) [inline package.json:145]: who is managing the source code of this library? why don't we put it in github.com/openforis/...?

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: webapp/components/form/DateInput/ReactDateInput/component.js [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+            moveNext,
+            dateKey: dateKeys[2],
+            onFocus: handleFocus,
+            onBlur: handleBlur,
+            showLabel,
+          })}
+        </div>
+        {error && (
+          <div className="error-container">
+            <small>Invalid date</small>
```
Comment (minotogna) [inline webapp/components/form/DateInput/ReactDateInput/component.js:318]: texts must be added to i18n

File: webapp/components/form/DateInput/ReactDateInput/component.js [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+          </div>
+        )}
+        {warning && (
+          <div className="error-container">
+            <small>Incomplete date</small>
```
Comment (minotogna) [inline webapp/components/form/DateInput/ReactDateInput/component.js:323]: same as above

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — DISMISSED** [review]

File: webapp/components/form/DateInput/ReactDateInput/component.js [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+    },
+    ref
+  ) => {
+    const i18n = useI18n()
+
+    const dateInputContainer = useRef(null)
+    const calendarContainer = useRef(null)
+    const [dateInputContainerRef, setDateInputContainerRef] = useState(null)
+    const [localDate, setDate] = useState('')
+    const [elRefs, setElRefs] = React.useState([])
```
Comment (SteRiccio) [inline webapp/components/form/DateInput/ReactDateInput/component.js:78]: use just `useState`

**Review by SteRiccio — APPROVED** [review]

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**